lib/ex_admin/resource_title_actions.ex

```diff
@@ -0,0 +1,85 @@
+defmodule ExAdmin.ResourceTitleActions do
+  alias ExAdmin.Utils
+  @doc false
+  def default(%Plug.Conn{params: params} = conn, %{resource_model: resource_model} = defn) do
+    singular = Utils.displayable_name_singular(conn) |> Utils.titleize
+    actions = defn.actions
+    case Utils.action_name(conn) do
+      :show ->
+        id = Map.get(params, "id")
+        Enum.reduce([:edit, :new, :delete], [], fn(action, acc) ->
+          if Utils.authorized_action?(conn, action, resource_model) do
+            [{action, action_button(conn, defn, singular, :show, action, actions, id)}|acc]
+          else
+            acc
+          end
+        end)
+        |> add_custom_actions(:show, actions, conn, resource_model, id)
+        |> Enum.reverse
+      action when action in [:index, :edit] ->
+        if Utils.authorized_action?(conn, action, resource_model) do
+          [{:new, action_button(conn, defn, singular, action, :new, actions)}]
+        else
+          []
+        end
+        |> add_custom_actions(action, actions, conn, resource_model)
+        |> Enum.reverse
+      _ ->
+        []
+    end
+  end
+
+  @doc false
+  def action_button(conn, defn, name, _page, action, actions, id \\ nil) do
```

Comment:
I know this is the old code, just moved. But I'd appreciate if you could remove the nesting. Something like this (not tested):

```elixir
  def action_button(conn, defn, name, _page, action, actions, id \\ nil) do
    with true <- action in actions, 
         true <- Utils.authorized_action?(conn, action, defn) do
      
      action_name = defn.action_labels[action] || Utils.humanize(action)
      [action_link(conn, "#{action_name} #{name}", action, id)]
    else
      _ -> []
    end
  end
```